build_tools/docker/cmake/Dockerfile

```diff
@@ -0,0 +1,59 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# An image for building IREE using CMake.
+
+# Build using:
+# docker build --tag gcr.io/iree-oss/cmake build_tools/docker/cmake/
+
+# Run using:
+# docker run -it --rm --entrypoint bash \
+#   --volume /home/phoenix/iree/fork:/usr/src/iree/ \
+#   gcr.io/iree-oss/cmake
```

Comment:
This starts the container interactively. Also probably good not to include your specific home directory :-D

```suggestion
# Run interactively using:
# docker run -it --rm --entrypoint bash \
#   --volume ${IREE_GIT_DIR?}:/usr/src/iree/ \
#   gcr.io/iree-oss/cmake
```

Reply:
I actually didn't intend for this to make it in, those were notes I had while testing 😅

Reply:
I'm going to remove this. Also, I haven't seen `IREE_GIT_DIR` in any of the documentation or code - is it something we use?

Reply:
It's just an env variable with a clear name. I like this format for "fill in here" because it means that the command is actually valid on its own and someone can set an env variable to use it (or replace the text if that's easier). Using a consistent env variable for it makes it better. I've seen lots of docs where the instructions include a series of command lines like `foo [PROJECT-ID]` and it drives me nuts because if all of those had been `"${PROJECT_ID?}"` I could've set the env variable once and then done the whole thing with just copy-paste, but instead I have to manually rewrite every single command (*cough* google cloud docs *cough*)